sphinx/screens/chats/chat-common/chat-common/src/main/java/chat/sphinx/chat_common/util/AudioRecorderController.kt

```diff
@@ -0,0 +1,90 @@
+package chat.sphinx.chat_common.util
+
+import android.media.MediaRecorder
+import androidx.navigation.NavArgs
+import chat.sphinx.chat_common.ui.ChatViewModel
+import chat.sphinx.chat_common.ui.viewstate.footer.FooterViewState
+import io.matthewnelson.concept_coroutines.CoroutineDispatchers
+import io.matthewnelson.concept_media_cache.MediaCacheHandler
+import kotlinx.coroutines.*
+import java.io.File
+
+class AudioRecorderController<ARGS : NavArgs>(
+    private val viewModel: ChatViewModel<ARGS>,
+    private val viewModelScope: CoroutineScope,
+    val mediaCacheHandler: MediaCacheHandler,
+    dispatchers: CoroutineDispatchers,
+): CoroutineDispatchers by dispatchers {
+    var recordingTempFile: File? = null
+    private var recorder: MediaRecorder? = null
```

Comment:
use a `Pair<File, MediaRecorder>?` to keep them together in the same local variable, and a getter value to return the file if available when queried by the chat fragment.